waiter/src/waiter/service_description.clj

```diff
@@ -98,23 +98,30 @@
                                         (map :k)
                                         (set)))
 
-; keys used in computing the service-id from the service description
 (def ^:const service-override-keys
   #{"authentication" "blacklist-on-503" "concurrency-level" "distribution-scheme" "expired-instance-restart-rate"
     "grace-period-secs" "health-check-interval-secs" "health-check-max-consecutive-failures"
     "idle-timeout-mins" "instance-expiry-mins" "interstitial-secs" "jitter-threshold" "max-queue-length" "min-instances"
     "max-instances" "restart-backoff-factor" "scale-down-factor" "scale-factor" "scale-up-factor"})
 
-; keys stored in the service description
-(def ^:const service-description-keys
-  (set/union service-override-keys
-             #{"allowed-params" "backend-proto" "cmd" "cmd-type" "cpus" "env" "health-check-url" "mem" "metadata"
-               "metric-group" "name" "permitted-user" "ports" "run-as-user" "version"}))
+(def ^:const service-non-override-keys
+  #{"allowed-params" "backend-proto" "cmd" "cmd-type" "cpus" "env" "health-check-url" "mem" "metadata"
+    "metric-group" "name" "permitted-user" "ports" "run-as-user" "version"})
+
+; keys used as parameters in the service description
+(def ^:const service-parameter-keys
+  (set/union service-override-keys service-non-override-keys))
+
+; keys allowed in service description metadata, these need to be distinct from service parameter keys
+(def ^:const service-metadata-keys #{"source-tokens"})
```

Comment:
How are these different from the "other" metadata keys?

Reply:
Currently all the fields we store in the service description are the service parameters, `surce-tokens` will be the first of its kind - a metadata entry that will be system generated, that we will not allow end users to configure.